src/util/http_utils.js

```diff
@@ -678,8 +678,8 @@ function authorize_session_token(req, options) {
 function validate_nsfs_whitelist(req) {
     // remove prefix for V4 IPs for whitelist validation
     const client_ip = parse_client_ip(req).replace(/^::ffff:/, '');
-    if (config.NSFS_WHITELIST.length === 0) return;
-    for (const whitelist_ip of config.NSFS_WHITELIST) {
+    if (config.S3_CLIENT_IP_WHITELIST.length === 0) return;
+    for (const whitelist_ip of config.S3_CLIENT_IP_WHITELIST) {
```

Comment:
Wait, aren't we doing the wrong thing here? this should actually be S3_SERVER_IP_WHITELIST ...

If I remember correctly, this whitelist is needed to restrict accepting requests to certain host interfaces, by comparing the server IP's.

But our code (from before this PR) seems to be checking the **client** IP (`req.connection.remoteAddress`).

So unless I am missing something, I think this needs fixing and re-validation ASAP.

@naveenpaul1 @romayalon @rkomandu @madhuthorat

Reply:
@guymguym I created a severity 1 issue - https://github.com/noobaa/noobaa-core/issues/7999
will handle it in a seperate PR